R/render.r

```diff
@@ -422,7 +422,94 @@ data_deps <- function(pkg, depth) {
       )
     )
   )
-  deps <- bslib::bs_theme_dependencies(bs_theme)
+
+  # map secondary to component-active-bg (thus also dropdown-link-active-bg)
+  # unless a value was set by the user
+  if (is.null(pkg$meta[["template"]]$bslib$`component-active-bg`)) {
+    bs_theme <- bslib::bs_add_variables(
+      bs_theme,
+      "component-active-bg" = as.character(
+        bslib::bs_get_variables(bs_theme, "secondary")
+      )
+    )
+  }
+
+  # map body color to navbar colors
+  if (is.null(pkg$meta[["template"]]$bslib$`navbar-light-color`)) {
+    bs_theme <- bslib::bs_add_variables(
+      bs_theme,
+      "navbar-light-color" = "rgba($body-color, 0.8)",
+      .where = "declarations"
+    )
+  }
+  if (is.null(pkg$meta[["template"]]$bslib$`navbar-light-hover-color`)) {
+    bs_theme <- bslib::bs_add_variables(
+      bs_theme,
+      "navbar-light-hover-color" = "rgba($body-color, 0.9)",
+      .where = "declarations"
+    )
+  }
+
+  # map primary to navbar-light-active-color + component-active-color
+  # unless a value was set by the user
+  if (is.null(pkg$meta[["template"]]$bslib$`navbar-light-active-color`)) {
+    bs_theme <- bslib::bs_add_variables(
+      bs_theme,
+      "navbar-light-active-color " = as.character(
```

Comment:
this is my current workaround to ensure variables inheriting from `$navbar-light-active-color` are also reset cf https://github.com/r-lib/pkgdown/issues/1698#issuecomment-847108662